Enable disabled ActivateFw and AuthAndStash integ tests

mlvisaya opened the pull request:
Previously all ActivateFw and AuthAndStash integ tests were disabled when testing FPGA.

This was because these integ tests needs an external test SRAM to exchange data with Caliptra. SW emulator is using a SW SRAM which is not available previously in HW.

To work around this, we will be using the MCU MBOX SRAM as a test SRAM to enable these tests.

mlvisaya committed:
Enable disabled ActivateFw and AuthAndStash integ tests

Previously all ActivateFw and AuthAndStash integ tests were disabled
when testing FPGA.

This was because these integ tests needs an external test SRAM to
exchange data with Caliptra. SW emulator is using a SW SRAM which is not
available previously in HW.

To work around this, we will be using the MCU MBOX SRAM as a test SRAM
to enable these tests.

mlvisaya committed:
fix compile error for fpga_realitime

swenson commented on .github/workflows/fpga-subsystem.yml:
```diff
@@ -297,6 +297,7 @@ jobs:
               -E 'package(caliptra-image-elf)'
               -E 'package(caliptra-image-types)'
               -E 'package(caliptra-lms-types)'
+              -E 'package(caliptra-runtime) & test(test_activate_firmware::test_activate_fw_id_not_in_manifest) - test(test_activate_firmware::test_activate_invalid_fw_id) - test(test_activate_firmware::test_activate_mcu_fw_success) - test(test_activate_firmware::test_activate_mcu_soc_fw_success) - test(test_activate_firmware::test_activate_soc_fw_success) - test(test_activate_firmware::test_inva ...
```
The `-` is subtracting the tests out I think?

mlvisaya replied:
yes you're right. i thought it was a separator. i'll correct it.

swenson commented on runtime/tests/runtime_integration_tests/test_authorize_and_stash.rs:
```diff
@@ -1008,19 +1099,28 @@ fn test_authorize_from_load_address() {
-    let mut model = set_auth_manifest_with_test_sram(Some(auth_manifest), &load_memory_contents);
-
+    };
+    let mcu_image = [0xAAu8; 256];
```
Could we change these to not all be the same byte? (We've had problems before where got the endianness wrong when feeding to the SHA accelerator, for instance.)

zhalvorsen commented on runtime/tests/runtime_integration_tests/test_activate_firmware.rs:
```diff
@@ -90,7 +100,38 @@ fn load_and_authorize_fw(images: &[Image]) -> DefaultHwModel {
+    {
+        println!("locking  MCU mailbox SRAMs");
+        unsafe {
+            let mcu_mbox_exec_ptr = model.mci.ptr.add(0x600018 / 4) as *mut u32;
+            mcu_mbox_exec_ptr.write_volatile(0x1);
+
+            let mcu_mbox_lock_ptr = model.mci.ptr.add(0x600000 / 4) as *mut u32;
+            let _ = mcu_mbox_lock_ptr.read_volatile();
```
@swenson what do you think of pulling the MCU manager into `caliptra-sw` so we can use all of the peripherals? I doubt this is the only place we would use them in this repo. I can do it in a follow on

zhalvorsen replied:
Wouldn't you want to write `0` to execute in that scenario so it clears any previously started command?

clundin25 commented on runtime/tests/runtime_integration_tests/test_activate_firmware.rs:
```diff
@@ -143,16 +221,12 @@ fn test_activate_mcu_fw_success() {
         .expect("We should have received a response");
 }
 
+#[cfg(not(feature = "fpga_realtime"))]
```
Hmm I think @swenson has been using the model to check for subsystem to exit early. I think that might be more future proof

mlvisaya commented on runtime/tests/runtime_integration_tests/test_activate_firmware.rs:
```diff
@@ -90,7 +100,38 @@ fn load_and_authorize_fw(images: &[Image]) -> DefaultHwModel {
+            let mcu_mbox_sram_ptr = model.mci.ptr.add(0x400000 / 4) as *mut u32;
+
+            for (count, chunk) in test_sram_contents.chunks(4).enumerate() {
+                let mut word = 0u32;
+                for (i, byte) in chunk.iter().enumerate() {
+                    word |= (*byte as u32) << (i * 8);
+                }
+                mcu_mbox_sram_ptr
+                    .offset(count as isize)
+                    .write_volatile(word);
```
good idea

mlvisaya committed:
Addressed feedback

- Currently CI FPGA bitstream is flaky in running runtime, disable these
tests for now. It did pass on local FPGA tests
- Use a single function to write to test sram
- correct yaml filters

mlvisaya committed:
Correct unlock procedure

mlvisaya committed:
Merge branch 'main-2.x' into mlvisaya/enable-activate-firmware-test2

mlvisaya committed:
Merge branch 'main-2.x' into mlvisaya/enable-activate-firmware-test2

mlvisaya committed:
Merge branch 'main-2.x' into mlvisaya/enable-activate-firmware-test2

swenson reviewed (approved):
LGTM
